examples/hyperpars_nb/script.jl

```diff
@@ -0,0 +1,344 @@
+    return term1 - term2 - term3
+end
+
+# This function had to be modified for the `NonCentered` `SparseVariationalApproximation`
+# This is an educated guess, hopefully it is correct!
+
+function cavi!(fz::AbstractGPs.FiniteGP, lik, x, y, m, S, qΩ; niter=10)
+    K = ApproximateGPs._chol_cov(fz)
+    κ = K.L
+    for _ in 1:niter
+        post_u = u_posterior(fz, m, S)
+        post_fs = marginals(post_u(x))
+        aux_posterior!(qΩ, lik, y, post_fs)
+        S .= inv(Symmetric(I + κ' * Diagonal(only(expected_auglik_precision(lik, qΩ, y))) * κ))
+        m .= S * κ' * only(expected_auglik_potential(lik, qΩ, y)) - κ \ mean(fz)
+    end
+    return m, S
+end
+
+function build_posterior(l::AVGPLoss, θ)
+    f = build_model(l, θ)
+    lik = f.lik
+    fz = f(l.x).fx
+    N = length(l.x)
+    qΩ = init_aux_posterior(lik, N)
+    cavi!(fz, lik, l.x, l.y, l.m, l.S, qΩ; niter=100)
+    post = u_posterior(fz, l.m, l.S)
+    posty = LatentGP(post, lik, 1e-6)
+    residuals = l.y - mean(posty(l.x))
+    return posty, residuals
+end
+
+
+# ## Overloads for posterior over Y
+# We can calculate the mean and variance of Y analytically
+function Statistics.mean(f::AbstractGPs.LatentFiniteGP{T1, NegativeBinomialLikelihood{NBParamFailure{T2}, LogisticLink}}) where {T1, T2}
+    r = f.lik.params.failures
+    m, c = mean(f.fx), var(f.fx)
+    return @. r * exp(m + c / 2)
+end
+
+function Statistics.var(f::AbstractGPs.LatentFiniteGP{T1, NegativeBinomialLikelihood{NBParamFailure{T2}, LogisticLink}}) where {T1, T2}
+    r = f.lik.params.failures
+    c = var(f.fx)
+    my = mean(f)
+    return @. my * (1 + my * ((1 + inv(r)) * exp(c) - 1))
+end
+
+# We approximate the posterior over Y by a truncated Gaussian distribution
+function AbstractGPs.marginals(
+    f::AbstractGPs.LatentFiniteGP{T1, NegativeBinomialLikelihood{NBParamFailure{T2}, LogisticLink}}
+) where {T1, T2}
+    m, c = mean(f), var(f)
+    return @. truncated(Normal(m, sqrt(c)), zero(eltype(m)), nothing)
+end
+
+
+# ## Experiments
+# We generate some synthetic data
+
+Random.seed!(1234)
+N = 100
+x = range(0, 1; length=N)
+μ, σ² = initial_μ_σ²_Poisson(100, 5000)
+k = σ² * with_lengthscale(SEKernel(), 0.1)
+r = 100.
+gp = GP(μ - log(r), k)
+lik = NegativeBinomialLikelihood(NBParamFailure(r))
+lgp = LatentGP(gp, lik, 1e-6)
+f, y = rand(lgp(x))
+
+# We plot the latent function and the observations
+using Plots
+p1 = scatter(x, y, label = "y", color = :black, markersize = 3, legend = :outerleft)
+p2 = scatter(x, f, label = "f", legend = :outerleft, color = :black, lw = 1)
+plot(p1, p2; layout=(2, 1), size = (1000, 400))
+
+# We define the model loss functions
+loss1 = DSVGPLoss(x, y)
+loss2 = AVGPLoss(x, y)
+
+# We call and time the `optimize` function to find optimal parameters
+# The augmented GP leads to a considerable speedup, at least for this
+# problem size.
+@time θ1 = optimize(loss1)
+@time θ2 = optimize(loss2)
+
+# We evaluate the loss function after training
+@show loss1(θ1)
+@show loss2(θ2)
```

Comment:
Instead of `@show` you should use the Literate API 
```suggestion
loss1(θ1), loss2(θ2)
```